app/feature/feature-terminate-insurance/src/main/kotlin/com/hedvig/android/feature/terminateinsurance/step/survey/TerminationSurveyDestination.kt

```diff
@@ -102,174 +133,332 @@ private fun TerminationSurveyScreen(
+                      Text(
+                        text = "$length/140",
+                        style = MaterialTheme.typography.titleSmall,
+                        color = MaterialTheme.colorScheme.onSurfaceVariant,
+                      )
+                    }
+                  }
                 }
+                Spacer(modifier = (Modifier.height(4.dp)))
               }
             }
-            Spacer(modifier = (Modifier.height(4.dp)))
           }
         }
       }
+      Spacer(Modifier.height(12.dp))
+      HedvigContainedButton(
+        stringResource(id = R.string.general_continue_button),
+        enabled = uiState.continueAllowed,
+        modifier = Modifier.padding(horizontal = 16.dp),
+        colors = ButtonDefaults.buttonColors(
+          disabledContainerColor = MaterialTheme.colorScheme.surface,
+          disabledContentColor = MaterialTheme.colorScheme.onSurfaceVariant,
+        ),
+        onClick = onContinueClick,
+        isLoading = uiState.isNavigationStepLoading,
+      )
+      Spacer(Modifier.height(16.dp))
+    }
+
+    AnimatedVisibility(
+      visible = uiState.showFullScreenEditText != null,
+      enter = fadeIn() + slideInVertically(initialOffsetY = { it / 2 }),
+      exit = fadeOut(),
+    ) {
+      val reason = uiState.showFullScreenEditText
+      if (reason != null) {
+        FullScreenEditableText(
+          reason.feedBack,
+          modifier = Modifier
+            .fillMaxSize()
+            .windowInsetsPadding(
+              WindowInsets.safeDrawing.only(WindowInsetsSides.Horizontal + WindowInsetsSides.Vertical),
+            ),
+          onCancelClick = onCloseFullScreenEditText,
+          onSaveClick = { newFeedback ->
+            changeFeedbackForReason(reason.surveyOption, newFeedback)
+          },
+        )
+      }
     }
-    Spacer(Modifier.height(12.dp))
-    HedvigContainedButton(
-      stringResource(id = R.string.general_continue_button),
-      enabled = uiState.continueAllowed,
-      modifier = Modifier.padding(horizontal = 16.dp),
-      colors = ButtonDefaults.buttonColors(
-        disabledContainerColor = MaterialTheme.colorScheme.surface,
-        disabledContentColor = MaterialTheme.colorScheme.onSurfaceVariant,
+  }
+}
+
+@Composable
+private fun FullScreenEditableText(
+  feedbackText: String?,
+  onSaveClick: (String?) -> Unit,
+  onCancelClick: () -> Unit,
+  modifier: Modifier = Modifier,
+) {
+  val focusRequester = remember { FocusRequester() }
+  val hint = stringResource(id = R.string.TERMINATION_SURVEY_FEEDBACK_HINT)
+  var textValue by remember {
+    mutableStateOf(
+      TextFieldValue(
+        text = feedbackText ?: "",
+        selection = TextRange((feedbackText ?: "").length),
       ),
-      onClick = onContinueClick,
-      isLoading = uiState.isNavigationStepLoading,
     )
-    Spacer(Modifier.height(16.dp))
+  }
+  LaunchedEffect(Unit) {
+    focusRequester.requestFocus()
+  }
+  Column(
```

Comment:
This column seems to only have one child, another column. This could be the surface instead I suppose. Then the child column wouldn't need the `weight(1f)` either.